### DISCUSSION TITLE
Is there a way to dump the fetched content?

### DISCUSSION THREAD

**Discussion by koote** [lestrrat-go/jwx#1058] [General]
I hit a JSON deserialization error `failed to unmarshal JWK set` inside `jwk.Fetch` when running in production environment, obviously the response body is not in JSON format, I've check the `/certs` endpoint and tried to download using browser, everything works fine on my local computer as well as the test environment, I cannot repo this in a non-production environment, also there is no way to dump the content of response body when calling `jwk.Fetch`, is there a plan to add this feature? Thanks.

**Comment by lestrrat** [comment]
There were no plans. Do you have an API in mind?

**Comment by anakinkinsey** [comment]
Hey @koote not necessarily something you'd want to do in prod. But you probably could pass down an http.Client with a custom transport that logs out the response body. I'm facing the same error as you and hope to solve it this way.

**Reply by lestrrat** [reply]
If the the transport thing doesn't work out and if you have an API that you want to see in jwx, please let me know

**Reply by anakinkinsey** [reply]
It worked great and I was able to solve my issue. One possible improvement would be able to pass a `ctx` all the way down to the `client.Do()` func. Some observability vendors use the `ctx` along with a custom round tripper to trace requests. Currently I don't think the `ctx` is passed down all the way

**Reply by anakinkinsey** [reply]
ah ignore that ^ it appears the latest version does pass the `ctx` down: https://github.com/lestrrat-go/jwx/blob/develop/v3/jwk/fetch.go#L92
